scalafmt-tests/src/test/resources/test/Enum.stat

```diff
@@ -0,0 +1,51 @@
+runner.dialect = dotty
+<<< enum
+enum Color {
+case Red, Green, Blue
+}
+>>>
+enum Color {
+  case Red, Green, Blue
+}
+<<< parameterized enum
+enum Color (val rgb: Int) {
+case Red   extends Color(0xff0000)
+case Green extends Color(0x00ff00)
+case Blue  extends Color(0x0000ff)
+}
+>>>
+enum Color (val rgb: Int) {
+  case Red extends Color(0xff0000)
+  case Green extends Color(0x00ff00)
+  case Blue extends Color(0x0000ff)
+}
+<<< user-defined members of enums
+enum Planet (mass: Double, radius: Double) {
+private final val G = 6.67300e-11
+def surfaceGravity = G * mass / (radius * radius)
+def surfaceWeight(otherMass: Double) = otherMass * surfaceGravity
+
+case Mercury extends Planet(3.303e+23, 2.4397e6)
```

Comment:
Can you add some additional `with` statements and see where it breaks and what indentation is used?

Reply:
This is the result for now:
```
case Mercury
    extends Planet(
      3.303e+23,
      2.4397e6
    ) with Alpha with Beta with Gamma with Delta with Epsilon
```

Reply:
yes, I'm going to add such a case. It doesn't work though yet - the changes that fixed the case with the opening parenthesis and parameter list don't apply to the above case.

Reply:
> by the way, can enum itself extend anything, or only the cases?

they can. Following code parses:
```
  enum A extends Alpha with Beta with Gamma with Delta with Epsilon {
    case B
  }
```

Reply:
Changes in `TokenOps` made the above case pass.

Reply:
> additionally, it would be good to have tests to see how this works with `binPack.parentConstructors`, specifically the Oneline and OnelineIfPrimaryOneline. might need adjusted implementation since for enum primary is the extends, since the case doesn't take parameters. we have plenty of examples of how to adjust config for a single test case.

Here is the example passing test case related to this setting:
```
<<<
binPack.parentConstructors = Never
maxColumn = 30
===
enum Foo(val x: Int) {
  case Bar extends Foo(1) with X with Z
}
>>>
enum Foo(val x: Int) {
  case Bar
      extends Foo(
        1
      ) with X with Z
}
```
Here we use `class` kw instead of `enum` and the following passes:
```
<<<
binPack.parentConstructors = Never
maxColumn = 30
===
enum Foo(val x: Int) {
  class Bar extends Foo(1) with X with Z
}
>>>
enum Foo(val x: Int) {
  class Bar
      extends Foo(1)
      with X
      with Z
}
```
I'm not sure if we expect the exact same result with the enum as with class? 

BTW there seems to be a mistake in the docs:
```
Parent constructors are C and D in
"class A extends B with C and D" // probably should be "class A extends B with C with D"
```
In addition - is not B also the parent constructor of A?